gap/congruences/conglatt.gi

```diff
@@ -7,111 +7,81 @@
 ##
 #############################################################################
 ##
-## This file contains functions for a lattice of congruences.
+## This file contains functions for a poset of congruences.
 ##
 ## When the congruences of a semigroup are computed, they form a lattice with
 ## respect to containment.  The information about how the congruences lie in
-## this lattice is stored in an IsCongruenceLattice object (a positional object
-## based on a list) and can be retrieved from this object with the following
-## methods
+## this lattice may be stored in an IsCongruencePoset object (a component object
+## based on a record) and can be retrieved from this object with the following
+## methods.
+##
+## Most of these methods simply redirect to the partial order defined by this
+## object, that is a list of lists describing the partial order of the
+## congruences with respect to containment.
 ##
-## Most of these methods simply redirect to the first entry of the object, the
-## actual list of lists of integers which defines the containments in the
-## lattice.  Component 2 in the list is the list of congruences, which is hidden
-## when this object is used.
-
-InstallMethod(CongruencesOfLattice,
-"for a congruence lattice",
-[IsCongruenceLattice],
-function(latt)
-  return latt![2];
-end);
 
 InstallMethod(\[\],
-"for a congruence lattice",
-[IsCongruenceLattice, IsObject],
-function(latt, x)
-  return latt![1][x];
+"for a congruence poset",
+[IsCongruencePoset, IsObject],
```

Comment:
Why is this for `IsObject`? Shouldn't it be for `IsPosInt`? and the description string is wrong, should say "for a congruence poset and pos. int." or similar.